### ISSUE TITLE
Game soft locked after Lewis the Frog tragically died in fire during conversation

### ISSUE THREAD

**Issue by comradeshook** [opennox/opennox#662]
As much as it pains me that it even happened, it must be reported: When i spoke to Lewis in the burning building in chapter 5, he immediately started moving around and hopped into a fire, which his poor amphibian body did not survive. As i wasn't fast enough to exit the conversation with him, this left me stuck with the dialog box on screen, unable to close it.

Curiously, the dialog box is not seen on the screenshot here, but this was the area.
![nox9](https://github.com/noxworld-dev/opennox/assets/8018945/35f66d9a-b586-410e-b6b5-5217500a5956)

Can be reproduced in this save:
[SAVE0001.zip](https://github.com/noxworld-dev/opennox/files/13989001/SAVE0001.zip)

I was able to exit to the menu with Alt-F4, but this caused #663.

Playing on Windows, GOG version, v1.9.0 alpha12.

**Comment by KILAH4716** [comment]
The pathfinding for Lewis here is actually SO stupid, Westwood got fed up, and instead of fixing the problem, they gave Lewis **infinite Invulnerability** after successfully completing the quest!!!!!!! This is hands down the most stressful quest in all of Nox! Can we fix the pathfinding in general please?!

**Comment by comradeshook** [comment]
Ahah, what the fuck Westwood :'D

**Comment by KILAH4716** [comment]
By the way, after extensive testing, Lewis will not and cannot move under any conditions until you talk to him. So the best way to do this quest is to douse all flames in the house first, THEN talk to him, and you won't run into this problem.

**Comment by comradeshook** [comment]
Yeah, that's also how i usually go about it, though i'll admit i had forgotten that he IMMEDIATELY starts hopping around like an idiot the instant you click him. :V
Allies trying to follow you in general has pretty iffy pathfinding honestly, Horvath is also kind of a dumbass at getting through doorways when you need him to, especially the first doorway between the cave and the swamp.
